components/malloc_size_of/lib.rs

```diff
@@ -782,3 +791,91 @@ impl MallocSizeOf for xml5ever::QualName {
         self.local.size_of(ops)
     }
 }
+
+impl MallocSizeOf for hyper::header::Headers {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let mut t = 0;
+        for x in self.iter() {
+           t += x.size_of(ops)
+        }
+        t
+    }
+}
+
+impl<'a> MallocSizeOf for hyper::header::HeaderView<'a> {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // Close enough?
+        self.name().size_of(ops) + self.value_string().size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::header::ContentType {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // size_of::<hyper::header::ContentType>
+        self.0.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Mime {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.0.size_of(ops) +
+        self.1.size_of(ops) +
+        self.2.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Attr {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        self.len()
```

Comment:
Let's match on self, return 0 for non-Ext variants, and call size_of on the string member of Ext.

Reply:
I've encountered the compiler complaining about matching on enum self.
```
error[E0507]: cannot move out of borrowed content
   --> components/malloc_size_of/lib.rs:832:13
    |
832 |             &hyper::mime::Attr::Ext(String) => self.size_of(ops),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^------^
    |             |                       |
    |             |                       hint: to prevent move, use `ref String` or `ref mut String`
    |             cannot move out of borrowed content
```
Glancing at the document, the enum does derive Clone, so I'm not sure why I'm getting this error. I'm using this now:

```
  829 impl MallocSizeOf for hyper::mime::Attr {
~ 830     fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
~ 831         let value = self.clone();
+ 832         match value {
+ 833             hyper::mime::Attr::Ext(String) => self.size_of(ops),
+ 834             _ => 0,
+ 835         }   
  836     }   
  837 }   
```
And it's throwing warnings: 
```
warning: unused variable: `String`
   --> components/malloc_size_of/lib.rs:833:36
    |
833 |             hyper::mime::Attr::Ext(String) => self.size_of(ops),
    |                                    ^^^^^^
    |
    = note: #[warn(unused_variables)] on by default
    = note: to avoid this warning, consider using `_String` instead

warning: variable `String` should have a snake case name such as `string`
   --> components/malloc_size_of/lib.rs:833:36
    |
833 |             hyper::mime::Attr::Ext(String) => self.size_of(ops),
    |                                    ^^^^^^
    |
    = note: #[warn(non_snake_case)] on by default
```
It appears that we're not matching on a type and the compiler thinks `String` inside container `Ext` is a variable?

Reply:
Yes, the names inside of enum variants in matches are variables. Calling `self.size_of` will recurse into the current function, not invoke it on the inner type. https://doc.rust-lang.org/book/second-edition/ch06-00-enums.html should clear this up for you.

Reply:
All cleared up, thank you!!